### PR TITLE
Add parser for Hong Kong data

### PR DESCRIPTION
**Pull request by attwad**
For #518

Also bumps python version from 3.6 to 3.8, no hurting doing that.

Retrieval invocation:

```
SOURCE_API_URL=http://localhost:3001/api sam local invoke "RetrievalFunction" -e retrieval/valid_scheduled_event.json --docker-network=host
Invoking retrieval.lambda_handler (python3.8)
Image was not found.
Building image..............................................................................................................................................................................................................................................................................................................................
Failed to download a new amazon/aws-sam-cli-emulation-image-python3.8:rapid-1.0.0 image. Invoking with the already downloaded image.
Mounting /Users/timothe/healthmap/healthmap-gdo-temp/ingestion/functions/.aws-sam/build/RetrievalFunction as /var/task:ro,delegated inside runtime container
START RequestId: fe3360e9-2400-166c-70de-94315a114cec Version: $LATEST
Retrieving service account credentials from s3://epid-ingestion/covid-19-map-277002-0943eeb6776b.json
Requesting source configuration from http://localhost:3001/api/sources/5f311a9795e338003016593a
Received source API response: {'_id': '5f311a9795e338003016593a', 'name': 'hk', 'origin': {'_id': '5f311a9795e338003016593b', 'url': 'http://www.chp.gov.hk/files/misc/enhanced_sur_covid_19_eng.csv'}, 'format': 'CSV', '__v': 0}
Downloading CSV content from http://www.chp.gov.hk/files/misc/enhanced_sur_covid_19_eng.csv
Uploaded source content to s3://epid-sources-raw/5f311a9795e338003016593a/2020/08/10/1009/content.csv
END RequestId: fe3360e9-2400-166c-70de-94315a114cec
REPORT RequestId: fe3360e9-2400-166c-70de-94315a114cec	Init Duration: 4388.10 ms	Duration: 6977.83 ms	Billed Duration: 7000 ms	Memory Size: 128 MB	Max Memory Used: 52 MB
```

Parsing invocation:

```
SOURCE_API_URL=http://localhost:3001/api sam local invoke "HongKongParsingFunction" -e  parsing/hongkong/input_event.json --docker-network=host
START RequestId: 25b1a33c-07fd-18eb-9c98-40e9edc74c51 Version: $LATEST
Retrieving raw data from s3://epid-sources-raw/5f311a9795e338003016593a/2020/08/10/1009/content.csv
Skipping probable case
Retrieving service account credentials from s3://epid-ingestion/covid-19-map-277002-0943eeb6776b.json
Sending 4079 cases to http://localhost:3001/api/cases/batchUpsert
END RequestId: 25b1a33c-07fd-18eb-9c98-40e9edc74c51
REPORT RequestId: 25b1a33c-07fd-18eb-9c98-40e9edc74c51	Init Duration: 1507.39 ms	Duration: 53969.89 ms	Billed Duration: 54000 ms	Memory Size: 128 MB	Max Memory Used: 50 MB
```

### CODE REVIEW

File: ingestion/functions/parsing/hongkong/hongkong.py [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+        for row in reader:
+            # CSV contains both "Probable" and "Confirmed" cases.
+            # We only ingest confirmed cases.
+            if row[_CERTAINTY_INDEX] != "Confirmed":
+                print('Skipping probable case')
+                continue
+            case = {
+                "caseReference": {
+                    "sourceId": source_id,
+                    "sourceEntryId": row[_CASE_ID_INDEX],
```
Comment (axmb) [inline ingestion/functions/parsing/hongkong/hongkong.py:80]: :clap:

File: ingestion/functions/template.yaml [hunk 2]
```diff
@@ -18,7 +18,7 @@ Resources:
     Properties:
       CodeUri: retrieval/
       Handler: retrieval.lambda_handler
-      Runtime: python3.6
+      Runtime: python3.8
```
Comment (axmb) [inline ingestion/functions/template.yaml:21]: This is fine (can't recall exactly why 3.6 was used initially) -- but need to update the README.

Comment (attwad) [reply ingestion/functions/template.yaml:21]: I updated the last references of it in the readme.

File: ingestion/functions/parsing/hongkong/hongkong.py [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+                    "value": "discharged",
+                })
+            elif row[_OUTCOME_INDEX] == "Deceased":
+                case["events"].append({
+                    "name": "outcome",
+                    "value": "deceased",
+                })
+            elif row[_OUTCOME_INDEX] == "Hospitalized":
+                case["events"].append({
+                    "name": "hospitalAdmission",
```
[the remark this answers is not in the document]
Comment (attwad) [reply ingestion/functions/parsing/hongkong/hongkong.py:131]: yup, _Yes_ I mean.

File: ingestion/functions/parsing/hongkong/hongkong.py [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+                    "dateRange": {
+                        "start": convert_date(row[_ONSET_INDEX]),
+                        "end": convert_date(row[_ONSET_INDEX]),
+                    }
+                })
+            # Status can be Hospitalised/Discharged/Deceased.
+            if row[_OUTCOME_INDEX] == "Discharged":
+                case["events"].append({
+                    "name": "outcome",
+                    "value": "discharged",
```
[the remark this answers is not in the document]
Comment (attwad) [reply ingestion/functions/parsing/hongkong/hongkong.py]: ah right, I still see examples in the data model tests with Death and Recovered, changed to those values.

File: ingestion/functions/parsing/hongkong/hongkong.py [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+                "notes": row[_NOTES_INDEX],
+            }
+            # If patient was symptomatic, the onset date is set otherwise they are marked
+            # as "Asymptomatic" in the date column in the CSV.
+            # There are other values as well such as "Pending" or "Unknown".
+            if row[_ONSET_INDEX] == "Asymptomatic":
+                case["symptoms"] = {
+                    "status": "Asymptomatic",
+                }
+            elif "/" in row[_ONSET_INDEX]:
```
[the remark this answers is not in the document]
Comment (attwad) [reply ingestion/functions/parsing/hongkong/hongkong.py:107]: I think that was implied by the fact that we had an onsetDate for symptoms but yeah adding it as well would make querying the data easier so added.

File: ingestion/functions/parsing/hongkong/hongkong.py [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+    return date.strftime("%m/%d/%YZ")
+
+def convert_gender(raw_gender: str):
+    if raw_gender.upper() == "M":
+        return "Male"
+    elif raw_gender.upper() == "F":
+        return "Female"
+    return None
+
+def convert_age(age: float):
```
[the remark this answers is not in the document]
Comment (attwad) [reply ingestion/functions/parsing/hongkong/hongkong.py:59]: yup, gender and age, always 100% in the current dataset.

File: ingestion/functions/parsing/hongkong/hongkong.py [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+_OUTCOME_INDEX = 6
+_NOTES_INDEX = 8
+_CERTAINTY_INDEX = 9
+
+def convert_date(raw_date: str):
+    """
+    Convert raw date field into a value interpretable by the dataserver.
+    """
+    # The date is inconsistently listed in dd/mm/YYYY format or in d/m/yy
+    # we pad the later to make it look the same and be parseable.
```
[the remark this answers is not in the document]
Comment (attwad) [reply ingestion/functions/parsing/hongkong/hongkong.py]: done

File: ingestion/functions/parsing/hongkong/hongkong.py [hunk 8]
```diff
@@ -0,0 +1,134 @@
...
+            os.path.dirname(os.path.dirname(os.path.abspath(__file__))),
+            'common/python'))
+import parsing_lib
+
+# Fixed location, all cases are for Hong Kong.
+_LOCATION = {
+    "country": "China", # "One country, two systems". We only store countries here.
+    "administrativeAreaLevel1": "Hong kong",
+    "geoResolution": "Admin1",
+    "name": "Hong kong",
```
[the remark this answers is not in the document]
Comment (attwad) [reply ingestion/functions/parsing/hongkong/hongkong.py]: done

**Pull request merged**